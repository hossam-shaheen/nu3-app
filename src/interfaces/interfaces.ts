export interface ProductType {
    id: number;
    title: string;
    price: number;
    image: string;
    vendor: string;
    link: string;
}

export interface ProductsType {
    totalItems: number;
    items: ProductType[];
}

export interface ProductsSearchProps {
    onSearch: (searchKeyWord: string) => void;
}

export interface SortProps {
    onSort: (sortKeyWord: string) => void;
}
export interface ImageType {
    id: number;
    src: string;
    alt: string;
}
export interface ProductAttributeProps {
    productImage: ImageType;
    productImages: ImageType[];
    ProductTitle: string;
    productBodyHtml: string;
    productVendor: string;
}


export interface ProductVariantsProps {
    options: OptionsType[];
    variants: VariantType[];
}


export interface ProductInfoType {
    id: number;
    title: string;
    body_html: string;
    vendor: string;
    images: ImageType[];
    image: ImageType;
    options: OptionsType[];
    variants: VariantType[];
    url: string;
}

export interface VariantType {
    id: number,
    title: string,
    price: number,
}

export interface OptionsType {
    id: number,
    values: string[],
    name: string,
}

export interface ErrorType {
    icon?: string,
    message: string
}