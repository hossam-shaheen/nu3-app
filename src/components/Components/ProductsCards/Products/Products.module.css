.products {
    display: flex;
    flex-wrap: wrap;
    max-width: 1200px;
}