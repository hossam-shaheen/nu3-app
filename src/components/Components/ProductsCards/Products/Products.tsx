import { FunctionComponent } from 'react';
import { ProductType } from 'src/interfaces/interfaces';
import Product from '../Product/Product';
import classes from './Products.module.css';

export const Products: FunctionComponent<{ products: ProductType[] }> = ({ products }): JSX.Element => {

    const renderProducts = (): JSX.Element[] => {
        return products.map((product: ProductType) => <Product key={product.id} product={product} />)
    }

    return <ul className={classes.products}>
        {products.length > 0 && renderProducts()}
    </ul>
}
export default Products;