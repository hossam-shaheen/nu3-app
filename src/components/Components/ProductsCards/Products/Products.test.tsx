import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ProductType } from 'src/interfaces/interfaces';
import Products from './Products';

export const mockedProducts: ProductType[] = [{
    id: 4725777367121,
    image: "https://cdn.shopify.com/s/files/1/0095/2274/1329/products/8_252F1_252Fc_252Fc_252F81ccf47420c1f3a11717ff5ad7f93e3e13214556_21446_large.jpg?v=1620826303",
    link: "/products/nu3-premium-skin-and-hair-beauty",
    price: 50.99,
    title: "nu3 Premium Skin & Hair Beauty",
    vendor: "nu3",
}];

test('Products pass Empty Array : Renders', () => {
    render(<Products products={[]} />);
    const productsCards = screen.queryAllByRole("listitem");
    expect(productsCards.length).toBe(0);
});


test('Products pass array of one product : Renders', () => {
    render(<BrowserRouter>
        <Products products={mockedProducts} />
    </BrowserRouter>);
    const productsCards = screen.queryAllByRole("listitem");
    expect(productsCards.length).toBe(1);
});


test('Products pass Empty Array : Snapshot', () => {
    const component = renderer.create(<Products products={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Products pass array of one product : Snapshot', () => {
    const component = renderer.create(<BrowserRouter>
        <Products products={mockedProducts} />
    </BrowserRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});