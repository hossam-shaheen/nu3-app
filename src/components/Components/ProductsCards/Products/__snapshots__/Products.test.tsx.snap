// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products pass Empty Array : Snapshot 1`] = `
<ul
  className="products"
/>
`;

exports[`Products pass array of one product : Snapshot 1`] = `
<ul
  className="products"
>
  <li
    className="product"
  >
    <div
      className="product-image"
    >
      <a
        href="/product/nu3-premium-skin-&-hair-beauty"
        onClick={[Function]}
      >
        <img
          alt="nu3 Premium Skin & Hair Beauty"
          src="https://cdn.shopify.com/s/files/1/0095/2274/1329/products/8_252F1_252Fc_252Fc_252F81ccf47420c1f3a11717ff5ad7f93e3e13214556_21446_large.jpg?v=1620826303"
        />
      </a>
    </div>
    <div
      className="product-content"
    >
      <h3>
        nu3 Premium Skin & Hair Beauty
      </h3>
      <p
        className="product-price"
      >
        â‚¬50.99
      </p>
      <button>
         
        <i
          className="fas fa-shopping-cart"
        />
         Add To Cart
      </button>
    </div>
  </li>
</ul>
`;
